src/main/java/com/google/api/generator/engine/ast/AssignmentExpr.java

```diff
@@ -59,6 +59,13 @@ public AssignmentExpr build() {
               String.format(
                   "LHS type %s must match RHS type %s", lhsType.toString(), rhsType.toString()));
         }
+      } else if (rhsType.isPrimitiveType()) {
```

Comment:
Given the `TypeNode.equals()` changes, this whole check can be reworked to be simpler. That is,
`if (lhsType.isPrimitiveType() || rhsType.isPrimitiveType())`, assert they are equal and they are not null. I'll leave the second half of this as an exercise for the reader. :)